test/rubygems/test_gem_package.rb

```diff
@@ -159,7 +159,11 @@ def test_add_files_symlink
     File.open 'lib/code.rb',  'w' do |io| io.write '# lib/code.rb'  end
 
     # NOTE: 'code.rb' is correct, because it's relative to lib/code_sym.rb
-    File.symlink('code.rb', 'lib/code_sym.rb')
+    begin
+      File.symlink('code.rb', 'lib/code_sym.rb')
+    rescue Errno::EACCES
+      skip "symlink - must be admin with no UAC on Windows"
```

Comment:
There should be a `if Gem.win_platform?`, otherwise this skip could show up on other platforms where this message does not make much sense to the user.

Reply:
If it hits the rescue and it's not windows, what should happen?  Nothing, raise the error, or something else?

I only added it because many windows desktops (including mine) will raise the error unless testing is run from an 'admin prompt/console'.  Servers are normally configured differently.  On Appvyor, the tests **do not skip**.

Reply:
I think that it should `raise` so that the exception propagates back up to the user.

Reply:
@colby

Rebased, and also added similar logic to line 514.  And yeah, I screwed up offsetting line 513...